readme.md

```diff
@@ -81,188 +75,137 @@ increaseAgeBy(2)
-1. [find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find)
-1. [every](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every)
-1. [some](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some)
+**Higher-Order Functions**
 
-Once in your groups...
+These functions are considered to be another key area -- perhaps the most important -- in the functional programming paradigm.
 
-1. Create an explain it to me like i'm 5 explanation
-1. Create a codepen / jsfiddle demoing how it works
-1. Come up with a way to remember what it is and what it does
+In functional programming languages, functions are values. That is, they can be stored in variables, or passed into other functions.
 
-## Recursion
+Higher-order functions are functions that can take functions as arguments and/or return a function as output. They are good for **function composition.** That is, taking a function and putting it into another allows us to compose a lot of smaller functions into a bigger function.
 
-<http://lmgtfy.com/?q=recursion>
+We've already seen an example of a higher-order function with `forEach`:
 
-Recursion is when a function calls itself.
+```js
+var fruits = ["apples", "bananas", "cherries"];
 
-Here's an infinite example:
+fruits.forEach(function(currentFruit) {
+  console.log("Every day I eat two " + currentFruit)
+});
 
-__Running this code in a browser will freeze your page__
+//vs
 
-```js
-function loop(){
-  loop()
+for(var i = 0; i < fruits.length; i++) {
+  console.log("Every day I eat two " + fruits[i]);
 }
-loop()
 ```
 
-The Fibonacci sequence is a classic example of recursion.
-
-The fibonacci numbers of `n` are the fibonacci numbers of `n - 1` + the
-fibonacci numbers of `n - 2`,
-where the fibonacci numbers of `1` and `2` are both 1.
+<details>
+  <summary>Anyone remember another one we've used quite frequently?</summary>
+  <p>`.on('click')`</p>
+</details>
 
-```
-3: 1 + 1  = 2
-4: 1 + 2  = 3
-5: 2 + 3  = 5
-6: 3 + 5  = 8
-7: 5 + 8  = 13
-```
+## Examples of functions that take functions as arguments:
 
-```js
-function fib(n){
-  if(n === 1 || n === 2){
-    return 1
-  }
-  return fib(n-1) + fib(n-2)
-}
-```
+### Filter
 
-### You do: Factorial
+ [`Filter`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) is a method on an array that accepts another function as it’s argument, which it will then use to return a new filtered version of the array.
 
-<https://en.wikipedia.org/wiki/Factorial>
 
-[Factorial](./02-factorial.html)
+```js
+var naysayers = [
+  {name: 'Adam', age: 320},
+  {name: 'Jared', age: 222},
+  {name: 'Will', age: 187},
+  {name: 'Isaac', age: 320},
+  {name: 'Rodney', age: 423},
+  {name: 'Fiona', age: 320},
+  {name: 'Mike', age: 239}
+]
 
-### Loops can be replaced with recursion
+naysayers.filter(function(naysayer){
+  return naysayer.age === 320;
+})
+```
```

Comment:
I think a better example would be to use a condition that returns multiple naysayers (e.g., `return naysayer.age > 300`). Otherwise you're using `filter` more like `find`.